src/ucp/api/ucp.h

```diff
@@ -538,6 +545,50 @@ typedef enum {
 } ucp_op_attr_t;
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief UCP AM receive data parameter fields and flags
+ *
+ * The enumeration allows specifying which fields in @ref ucp_am_recv_param_t
+ * are present and receive operation flags are used. It is used to enable
+ * backward compatibility support.
+ */
+typedef enum {
+    UCP_AM_RECV_ATTR_FIELD_REPLY_EP    = UCS_BIT(0),  /**< reply_ep field */
+
+    /**
+     * Indicates that the data provided in @ref ucp_am_data_callback_t callback
+     * can be held by the user. If UCS_INPROGRESS is returned from the callback,
+     * the data parameter will persist and the user has to call
+     * @ref ucp_am_data_release when data is no longer needed.
+     */
+    UCP_AM_RECV_ATTR_FLAG_DATA         = UCS_BIT(16),
```

Comment:
Any reason why the 17th bit is set for this flag?

Reply:
@Akshay-Venkatesh agree, since Spec is generated from header file it does not belong to the header. I think in NEWS section of the release we can list APIs that for now do not support Cuda memory.